### PR TITLE
Projects Home Page

### PR DESCRIPTION
**Pull request by daliamaarek**
In this PR: 
* Projects Home Page (Grid View) 
* Projects Search Bar
* Projects Sort Lexicographically  

| Initial Push | Update| Update 2 😀  |
| --- | --- | --- |
| Screen Shot 2020-04-13 at 7 28 16 PM | Screen Shot 2020-04-13 at 9 54 54 PM | Screen Shot 2020-04-15 at 9 31 18 PM |

### CODE REVIEW

File: frontend-zs/src/App.js [hunk 1]
```diff
@@ -39,6 +40,7 @@ class App extends React.Component {
             <PrivateRoute path="/editProfile" exact component={EditProfile} />
             <Route path='/login' component={Login} />
             <Route path='/signup' component={Signup} />
+            <Route path='/mitmachen' component={ProjectsList} />
```
Comment (ahmad-diaa) [inline frontend-zs/src/App.js]: Why is this path in German while the others are in English?

File: frontend-zs/src/components/projects/Home/Main.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import Axios from 'axios';
+import { Container, Row } from 'react-bootstrap'
+import { ProjectCardsList } from './ProjectCard';
+import { SearchBar } from './SearchBar';
+
+export function ProjectsList() {
+    const [allProjects, setAllProjects] = useState([])
+
```
Comment (ahmad-diaa) [inline frontend-zs/src/components/projects/Home/Main.js]: nit: too many blank lines

File: frontend-zs/src/components/projects/Home/ProjectCard.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+function goToProject(project) {
+    return <Redirect to={'/projects' + project._id} />
+
+}
+
+function ProjectCard(project, index) {
+    return (
+        <Row key={index}>
+            <Card bg="dark" text="white" className="text-left mt-md-3 mr-md-2 ml-md-2" tag="a"
```
Comment (klarmannsonia) [inline frontend-zs/src/components/projects/Home/ProjectCard.js]: I would propose to use their design guidline for the background color of the card. Maybe the dark grey color: #5F696A

File: frontend-zs/src/components/projects/Home/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        <Row middle="md" style={{ width: "100%" }} >
+            <Col xs={11}>
+                <input
+                    type="text"
+                    className="form-control mt-md-3 mb-md-3 mr-md-3"
+                    placeholder="Suche nach Titel"
+                    onChange={(evt) => onSearch(evt.target.value)} />
+            </Col>
+            <Col xs={1} onClick={() => sort()} className="pt-md-3" >
+                {sortAsc ? <FaSortAlphaUp /> : <FaSortAlphaDown />}
```
Comment (klarmannsonia) [inline frontend-zs/src/components/projects/Home/SearchBar.js]: Sort icon should be bigger.

**Review by klarmannsonia — APPROVED** [review]

**Review by ahmad-diaa — CHANGES_REQUESTED** [review]
Button to switch view is missing

**Review by ahmad-diaa — CHANGES_REQUESTED** [review]
Grid is not aligned with the search bar start

File: frontend-zs/src/components/projects/Home/SearchBar.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    return (
+        <Row middle='md' style={{ width: '100%' }} >
+            <Col xs={11}>
+                <input
+                    type='text'
+                    className='form-control mt-md-3 mb-md-3 mr-md-3'
+                    placeholder='Suche nach Titel'
+                    onChange={(evt) => onSearch(evt.target.value)} />
+            </Col>
+            <Col xs={1} onClick={() => sort()} className='pt-md-3' >
```
Comment (ahmad-diaa) [inline frontend-zs/src/components/projects/Home/SearchBar.js]: can't we decouple the sort item from the search bar?

File: frontend-zs/src/components/projects/Home/Main.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import Axios from 'axios';
+import { Container, Row } from 'react-bootstrap'
+import { ProjectCardsList } from './ProjectCard';
+import { SearchBar } from './SearchBar';
+
+export function ProjectsList() {
+    const [allProjects, setAllProjects] = useState([])
+    const [projects, setProjects] = useState(allProjects)
```
Comment (ahmad-diaa) [inline frontend-zs/src/components/projects/Home/Main.js]:
```suggestion
    const [displayProjects, setDisplayProjects] = useState(allProjects)
```

displayProjects might be more descriptive for the state. What do you think?

Comment (daliamaarek) [reply frontend-zs/src/components/projects/Home/Main.js:9]: Yup, good point. I'll change it

Comment (daliamaarek) [reply frontend-zs/src/components/projects/Home/Main.js:9]: Good point, I'll change it

**Review by ahmad-diaa — APPROVED** [review]

**Pull request merged**